src/odemis/driver/orsay.py

```diff
@@ -2760,3 +2771,242 @@ def terminate(self):
+                                                           )
+
+        self._positionYConnector = OrsayParameterConnector(self._positionY,
+                                                           self._hybridAperture.YPosition
+                                                           )
+
+        self._referencedXConnector = OrsayParameterConnector(self._referencedX,
+                                                             self._hybridAperture.Calibrated
+                                                             )
+
+        self._referencedYConnector = OrsayParameterConnector(self._referencedY,
+                                                             self._hybridAperture.Calibrated
+                                                             )
+
+        for aprtr_nmbr in range(NMBR_APERTURES):
+            connector_refs = self._apertureConnectors[f"Aperture{aprtr_nmbr}"]
+            connector_refs.append(OrsayParameterConnector(self._apertureDict[f"Aperture{aprtr_nmbr}"]["LifeTime"],
+                                                          recursive_getattr(self._hybridAperture,
+                                                                            f"Aperture{aprtr_nmbr}.Lifetime"))
+                                  )
+            connector_refs.append(OrsayParameterConnector(self._apertureDict[f"Aperture{aprtr_nmbr}"]["Size"],
+                                                          recursive_getattr(self._hybridAperture,
+                                                                            f"Aperture{aprtr_nmbr}.Size"))
+                                  )
+            connector_refs.append(OrsayParameterConnector(self._apertureDict[f"Aperture{aprtr_nmbr}"]["PositionX"],
+                                                          recursive_getattr(self._hybridAperture,
+                                                                            f"Aperture{aprtr_nmbr}.PositionX"))
+                                  )
+            connector_refs.append(OrsayParameterConnector(self._apertureDict[f"Aperture{aprtr_nmbr}"]["PositionY"],
+                                                          recursive_getattr(self._hybridAperture,
+                                                                            f"Aperture{aprtr_nmbr}.PositionY"))
+                                  )
+
+        self.update_VAs()
+        self._metadata[model.MD_APERTURES_INFO] = self._getAllApertureMetadata()
+
+    def selectAperture(self, id):
+        return self._executor.submit(self._selectAperture, id)
+
+    def _selectAperture(self, id, timeout=10):
+        logging.debug(f"Aperture {id} is selected")
+        tend = time.time() + timeout
+        self._hybridAperture.SelectedDiaph.Target = id
+        while int(self._hybridAperture.SelectedDiaph.Actual) != id \
+                and self._hybridAperture.IsMovingX.Actual == "True" \
+                and self._hybridAperture.IsMovingY.Actual == "True":
```

Comment:
Why use a `and` here? As soon as one axis stops, it'd be considered completed. I think you should use `or`, like in `_doMoveAbs()`.